src/analyzing/CFG_build.ml

```diff
@@ -47,15 +47,33 @@ type label_key = string * G.sid
  *)
 type state = {
   g : (F.node, F.edge) Ograph_extended.ograph_mutable;
-  (* When there is a 'return' we need to know the exit node to link to *)
-  exiti : F.nodei;
   (* Attaches labels to nodes. *)
   labels : (label_key, F.nodei) Hashtbl.t;
   (* Gotos pending to be resolved, a list of Goto nodes and the label
    * to which they are jumping. *)
   gotos : (nodei * label_key) list ref;
-  (* If we are inside a Try, this is the start of the handlers. *)
-  try_catches_opt : F.nodei option;
+  (* Destination node that a return node should go to. *)
+  return_destination : F.nodei;
+  (* True when a return may occur in any of the nodes being visited.
+   * Initialized as false, and once set to true, remains true forever.
+   * This field helps us determine whether we should propagate the return
```

Comment:
Could you clarify which situation is this describing?
> This field helps us determine whether we should propagate the return or not after executing a finally clause in exception handling.

Reply:
Hmm, OK, is this to make sure that a `return` inside a `try` will execute the `finally` clause before returning? IMO this deserved a separate PR (but let it be here for now).

Reply:
I'm removing return-related things in this PR. I'll resolve this.